Source/Core/InputCommon/ControllerInterface/WGInput/WGInput.cpp

```diff
@@ -668,6 +684,36 @@ static void RemoveDevice(const WGI::RawGameController& raw_game_controller)
 // (H is the lambda)
 #pragma warning(disable : 4265)
 
+static void HandleAddRemoveEvent(AddRemoveEvent evt)
```

Comment:
Nothing should block until completion here here. `WGI::RawGameController::RawGameControllerAdded()` registers a lambda to be called every time a new `RawGameController` is noticed by WGI. `s_device_add_remove_queue.EmplaceItem()` registers an item to be executed in the `WorkQueueThread`'s worker thread and does not wait for that item to be processed or anything like that -- it may block very shortly while the item is added in the queue if another thread is also adding to the queue or while the worker thread is taking an item from the queue to to be executed, but that can never cause a deadlock.